UnitystationLauncher/Models/Server.cs

```diff
@@ -28,15 +28,25 @@ public Server(string forkName, int buildVersion, string serverIp, int serverPort
         public string? LinuxDownload { get; set; }
         public (string, int) ForkAndVersion => (ForkName, BuildVersion);
 
-        public string? DownloadUrl =>
-            RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? WinDownload :
-            RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? OsxDownload :
-            RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? LinuxDownload :
-            throw new InvalidOperationException("Failed to detect OS");
+        public string? DownloadUrl
+        {
+            get
+            {
+                const string cdnUrl = "https://unitystationfile.b-cdn.net/";
+
+                var cdnPath = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? WinDownload :
+                    RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? OsxDownload :
+                    RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? LinuxDownload :
+                    throw new InvalidOperationException("Failed to detect OS");
+
+                return $"{cdnUrl}{cdnPath}";
```

Comment:
Wouldn't this break for all existing servers that use a full URL instead of a partial one?
Also, it doesn't really fix the issue since if the `cdnPath` points to a non-existent URL it will still not work.
If what you want is just locking downloads to our CDN then I would parse it as a `URI` and ensure the domain and maybe the protocol.
If you want to ensure it's valid then you need to test if the file exists with an HTTP Request.

Reply:
I will try parsing and rejecting non-compliants to appear in the list.